### ISSUE TITLE
Design decision needed: potentially unexpected package updates

### ISSUE THREAD

**Issue by gabalafou** [Quansight/conda-store#266]
tl;dr-- Current behavior: base packages will get updated whenever the user modifies their environment if the maintainers of those base packages released any newer versions. For the end user, is this good or bad? Is there a way we can mitigate any potential surprises via UI design?

Longer discussion copied from Slack.

Thread
[conda-store](https://quansight.slack.com/archives/C0291GQUX8T/p1647462418063949)

[Gabriel (UTC-6)](https://app.slack.com/team/U02JA56KDDY) [Today at 3:26 PM](https://quansight.slack.com/archives/C0291GQUX8T/p1647462418063949)
In my feature branch, when we create a new environment, the dependencies are specified as python and ipykernel  : they are not pinned to any version. The way my code currently works, that means that any time a Gator user adds, deletes, or updates any package, if python or ipykernel have since been updated, then they will get updated (even if the user isn’t specifically requesting to update python or ipykernel)
3 replies

[Gabriel (UTC-6)](https://app.slack.com/team/U02JA56KDDY)  [2 hours ago](https://quansight.slack.com/archives/C0291GQUX8T/p1647462427229449?thread_ts=1647462418.063949&cid=C0291GQUX8T)
Is that okay?

[Gabriel (UTC-6)](https://app.slack.com/team/U02JA56KDDY)  [2 hours ago](https://quansight.slack.com/archives/C0291GQUX8T/p1647462984613729?thread_ts=1647462418.063949&cid=C0291GQUX8T)
Imagine that on April 9, 2021, somebody creates an environment. Their environment.yaml file will look like:
```
name: foo
dependencies:
- python
- ipykernel
```
On April 9, the user will have Python 3.9 in their environment.
The next day, on April 10, Python is updated from 3.9 to 3.10.
On April 11, the user installs Node 16. My code sends a new specification to update the environment, by simply adding nodejs to the list of dependencies, like so:
```
name: foo
dependencies:
- python
- ipykernel
- nodejs=16
```
I believe what happens now is that when Conda Store triggers a new build to install Nodejs 16, it will also update python (because it wasn’t pinned in the specification file). So on April 11, the user will now have Python 3.10, without ever explicitly having asked to updated Python.
Is that expected behavior? [@Chris Ostrouchov](https://quansight.slack.com/team/UDL3C3CN9)

[Chris Ostrouchov](https://app.slack.com/team/UDL3C3CN9)  [2 hours ago](https://quansight.slack.com/archives/C0291GQUX8T/p1647463291816959?thread_ts=1647462418.063949&cid=C0291GQUX8T)
[@Gabriel (UTC-6)](https://quansight.slack.com/team/U02JA56KDDY) yes this is expected behavior. Though something to discuss if we would rather pin. To me pinning is generally not what users want. They really just want a stable environment which conda-store provides and does not change.

**Comment by kcpevey** [comment]
If you are using a package in a shared namespace, there is an implicit understanding that it **might** change. `With great power comes great responsibility`. Users need to be aware of how changing a shared environment might affect others, but ultimately if you want to be sure that you're environment will NOT change, then you should copy the env into your personal namespace, where it will be safe for all time.
